pkg/services/ldap/service/ldap.go

```diff
@@ -0,0 +1,88 @@
+package service
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/ldap"
+	"github.com/grafana/grafana/pkg/services/ldap/multildap"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+var (
+	ErrUnableToCreateLDAPClient = errors.New("unable to create LDAP client")
+)
+
+// LDAP is the interface for the LDAP service.
+type LDAP interface {
+	ReloadConfig() error
+	Config() *ldap.Config
+	Client() multildap.IMultiLDAP
+}
+
+type LDAPImpl struct {
+	client  multildap.IMultiLDAP
+	cfg     *setting.Cfg
+	ldapCfg *ldap.Config
+	log     log.Logger
+
+	// loadingMutex locks the reading of the config so multiple requests for reloading are sequential.
+	loadingMutex *sync.Mutex
+}
+
+func ProvideService(cfg *setting.Cfg) *LDAPImpl {
+	s := &LDAPImpl{
```

Comment:
Why does it necessary to instantiate `LDAPImpl` regardless of the LDAPEnabled setting (#44)?